nengo_spinnaker/simulator.py

```diff
@@ -278,10 +290,18 @@ def func(t, x, f=old_func):
 
     def close(self):
         """Clean the SpiNNaker board and prevent further simulation."""
-        # Stop the application
-        self._closed = True
-        self.io_controller.close()
-        self.controller.send_signal("stop")
+        if not self._closed:
+            # Stop the application
+            self._closed = True
+            self.io_controller.close()
+            self.controller.send_signal("stop")
 
     def trange(self, dt=None):
         return np.arange(1, self.steps + 1) * (self.dt or dt)
+
+
+def _weakclose(ref):
+    """Calls close on a simulator if it still exists."""
+    obj = ref()
+    if obj is not None:
+        obj.close()
```

Comment:
Hmm...  I might be misreading what's happening here, but it looks to me like if I create a Simulator and then forget my reference to it without closing, then it'll never get closed.